superset/assets/src/explore/components/PropertiesModal.jsx

```diff
@@ -0,0 +1,241 @@
+  useEffect(() => {
+    SupersetClient.get({
+      endpoint: `/api/v1/chart/related/owners`,
+    }).then(res => {
+      setUserOptions(
+        res.json.result.map(item => ({
+          value: item.value,
+          label: item.text,
+        })),
+      );
+    });
+  }, []);
+
+  const onSubmit = async event => {
+    event.stopPropagation();
+    event.preventDefault();
+    setSubmitting(true);
+    const payload = {
+      slice_name: name || null,
+      description: description || null,
+      cache_timeout: cacheTimeout || null,
+      owners: owners.map(o => o.value),
+    };
+    try {
+      const res = await SupersetClient.put({
+        endpoint: `/api/v1/chart/${slice.slice_id}`,
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(payload),
+      });
+      // update the redux state
+      onSave(res.json.result);
+      onHide();
+    } catch (res) {
+      const errObj = await getClientErrorObject(res);
+      showError(errObj);
+    }
+    setSubmitting(false);
+  };
+
+  return (
+    <form onSubmit={onSubmit}>
+      <Modal.Header closeButton>
+        <Modal.Title>Edit Chart Properties</Modal.Title>
+      </Modal.Header>
+      <Modal.Body>
+        <Row>
+          <Col md={6}>
+            <h3>{t('Basic Information')}</h3>
+            <FormGroup>
+              <label className="control-label" htmlFor="name">
+                {t('Name')}
+              </label>
+              <FormControl
+                name="name"
+                type="text"
+                bsSize="sm"
+                value={name}
+                onChange={event => setName(event.target.value)}
+              />
+            </FormGroup>
+            <FormGroup>
+              <label className="control-label" htmlFor="description">
+                {t('Description')}
+              </label>
+              <FormControl
+                name="description"
+                type="text"
+                componentClass="textarea"
+                bsSize="sm"
+                value={description}
+                onChange={event => setDescription(event.target.value)}
+                style={{ maxWidth: '100%' }}
+              />
+              <p className="help-block">
+                {t(
+                  'The description can be displayed as widget headers in the dashboard view. Supports markdown.',
+                )}
+              </p>
+            </FormGroup>
+          </Col>
+          <Col md={6}>
+            <h3>{t('Configuration')}</h3>
+            <FormGroup>
+              <label className="control-label" htmlFor="cacheTimeout">
+                {t('Cache Timeout')}
+              </label>
+              <FormControl
+                name="cacheTimeout"
+                type="text"
+                bsSize="sm"
+                value={cacheTimeout}
+                onChange={event =>
+                  setCacheTimeout(event.target.value.replace(/[^0-9]/, ''))
```

Comment:
I think if you made the `FormControl`'s input type `"number"` then you could get rid of this logic

Reply:
That's what I did at first, but I wasn't sure if we wanted to allow non-integers. Do you have any insight into whether that would be ok?

Reply:
The marshmallow schema on the endpoint wants an integer

Reply:
I suppose that when sending the request, you could do a `parseInt` on the value to remove the decimal. Maybe that's sufficient then? At least we don't need to run the validator onChange then

Reply:
`parseInt` can return `NaN` for certain inputs. An input of `type="number"` allows using "e" and decimal points. I'd prefer not to open the `NaN` door, personally.

Reply:
We could use `type="number` alongside the regex filter to get the extra little browser widgets that come with number inputs.